0000-template-protocol.md

```diff
@@ -0,0 +1,249 @@
+arrived at because of the problem. This helps other participants
+to react to errors with confidence. Formal state names are also used in the
+agent test suite, in log messages, and so forth.
+
+By convention, state names use lower-kebab-case but are compared
+case-insensitively and ignoring punctuation.
+
+State management in protocols is a deep topic. For more information, please
+see [State Details and State Machines](https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0003-protocols/state-details.md).
+
+### Messages
+
+If there is a message family associated with this protocol, this
+section describes each member of it. It should also note the names and
+versions of messages from other message families that are used by the
+protocol (e.g., an [`ack`](https://github.com/hyperledger/aries-rfcs/tree/master/features/0015-acks)
+or a [`problem-report`](https://github.com/hyperledger/aries-rfcs/tree/master/features/0035-report-problem)).
+Typically this section is written as a narrative, showing each message
+type in the context of an end-to-end sample interaction. All possible
+fields may not appear; an exhaustive catalog is saved for the "Reference"
+section.
+
+Sample messages that are presented in the narrative should also be checked
+in next to the markdown of the RFC, in [DIDComm Plaintext format](
+https://github.com/hyperledger/aries-rfcs/tree/master/features/0044-didcomm-file-and-mime-types#didcomm-messages-dm).
+
+#### Adoption Messages
+
+Adoption should be declared in an "Adopted" subsection of "Messages" in
+a protocol RFC. When adoption is specified, it should include a __minimum
+adopted version__ of the adopted message type: "This protocol adopts
+`ack` with version >= 1.4". All versions of the adopted message that share
+the same major number should be compatible, given the [semver rules](https://github.com/hyperledger/aries-rfcs/blob/master/concepts/0003-protocols/semver.md)
+that apply to protocols.
+
+### Constraints
+
+Many protocols have constraints that help parties build trust.
+For example, in buying a house, the protocol includes such things as
+commission paid to realtors to guarantee their incentives, title insurance,
+earnest money, and a phase of the process where a home inspection takes
+place. If you are documenting a protocol that has attributes like
+these, explain them here. If not, the section can be omitted.
+
+## Reference
+
+### Messages
+
+Unless the "Messages" section under "Tutorial" covered everything that
+needs to be known about all message fields, this is where the data type,
+validation rules, and semantics of each field in each message type are
+details. Enumerating possible values, or providing ABNF or regexes is
+encouraged. Following conventions such as [those for date-
+and time-related fields](https://github.com/dhh1128/indy-hipe/blob/dd3005803decfb87c13117bc7566a4fc90e5fa16/text/0027-date-time-conventions/README.md)
```

Comment:
This URI can be replaced with concepts/0074-didcomm-best-practices#date-time-conventions

Reply:
Changed